onnxruntime/contrib_ops/cpu/optional_ops/optional_ops.cc

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "optional_ops.h"
+#include "core/framework/ml_value.h"
+#include "core/providers/cpu/tensor/utils.h"
+
+namespace onnxruntime {
+namespace contrib {
+
+ONNX_OPERATOR_KERNEL_EX(Optional, kMSDomain, 1, kCpuExecutionProvider,
+                        KernelDefBuilder()
+                            .TypeConstraint("V", DataTypeImpl::AllTensorAndSequenceTensorTypes())
+                            .TypeConstraint("O", DataTypeImpl::AllOptionalTypes())
+                            // We may be able to re-use the input for the output as is unless the output
+                            // is a graph output. We provide this hint to the allocation planner
+                            // to make the re-use call.
+                            .Alias(0, 0),
```

Comment:
Wherever possible we should just re-use the input OrtValue as the output OrtValue - this op is essentially just a cast for the ONNX type - our runtime needn't duplicate the OrtValue. Of course, for graph outputs, we need to do a copy